gprofiler/java.py

```diff
@@ -73,7 +74,9 @@ def __init__(self, process: Process, storage_dir: str):
         # multiple times into the process)
         # without depending on storage_dir here, we maintain the same path even if gProfiler is re-run,
         # because storage_dir changes between runs.
-        self._ap_dir = os.path.join(TEMPORARY_STORAGE_PATH, "async-profiler")
+        # we embed the gprofiler version in the path, so different gprofiler versions can use different AP
+        # versions.
+        self._ap_dir = os.path.join(TEMPORARY_STORAGE_PATH, f"async-profiler-{__version__}")
```

Comment:
Perhaps give it a better name so it's not implied to be AP's version itself here

Reply:
Now it's really AP version